sdk/communication/azure-communication-chat/tests/_shared/test_communication_identity_serializer.py

```diff
@@ -107,25 +121,33 @@ def test_deserialize_phone_number(self):
 
     def test_serialize_teams_user(self):
         teams_user_identifier_model = CommunicationUserIdentifierSerializer.serialize(
-            MicrosoftTeamsUserIdentifier("teamsid")
+            MicrosoftTeamsUserIdentifier("teamsid", cloud=CommunicationCloudEnvironment.Public)
         )
 
         assert teams_user_identifier_model.kind is CommunicationIdentifierKind.MicrosoftTeamsUser
         assert teams_user_identifier_model.id is "teamsid"
+        assert teams_user_identifier_model.communication_cloud_environment is CommunicationCloudEnvironment.Public
 
     def test_deserialize_teams_user(self):
         teams_user_identifier_actual = CommunicationUserIdentifierSerializer.deserialize(
             CommunicationIdentifierModel(
                 kind=CommunicationIdentifierKind.MicrosoftTeamsUser,
                 microsoft_teams_user_id="teamsid",
-                is_anonymous=True
+                is_anonymous=True,
+                communication_cloud_environment="PUBLIC"
```

Comment:
id is not set, why does this test pass? deserialize should throw if id is missing. Is this test working correctly? https://github.com/Azure/azure-sdk-for-python/pull/16359/files#diff-6568c9c72c24dd2445fc151ffb9b623353622df55caa1c5a75b4eae453f01affR49

Reply:
Thanks for this. It seems that the model class for Teams has been missing the id field. I will make sure add it and rewrite the tests for this scenario.